src/components/marker-chart/Canvas.js

```diff
@@ -179,20 +188,12 @@ class MarkerChartCanvas extends React.PureComponent<Props, State> {
         continue;
       }
 
-      // Decide which samples to actually draw
-      const timeAtViewportLeft: Milliseconds =
-        rangeStart + rangeLength * viewportLeft;
-      const timeAtViewportRightPlusMargin: Milliseconds =
-        rangeStart +
-        rangeLength * viewportRight +
-        // This represents the amount of seconds in the right margin:
-        marginRight * (viewportLength * rangeLength / markerContainerWidth);
-
       let hoveredElement: MarkerDrawingInformation | null = null;
+      let previousDrawnPixel: number = -1;
       for (let i = 0; i < markerTiming.length; i++) {
         // Only draw samples that are in bounds.
         if (
-          markerTiming.end[i] > timeAtViewportLeft &&
+          markerTiming.end[i] >= timeAtViewportLeft &&
```

Comment:
This was leaving out dot markers at the start of the range -- I think it was incorrect _and_ it made the tests more difficult to read.